src/items.rs

```diff
@@ -88,49 +146,87 @@ impl<'a> Key<'a> {
     }
 }
 
+/// An item within a TOML document.
 #[derive(Debug, Clone, PartialEq)]
 pub enum Item<'a> {
+    /// A whitespace literal.
     WS(&'a str),
+    /// A comment literal.
     Comment(Trivia<'a>),
+    /// An integer literal.
     Integer {
+        /// The value of the integer.
         val: i64,
+        /// Trivia for the integer.
         meta: Trivia<'a>,
+        /// The original representation of the integer value.
         raw: &'a str,
     },
+    /// A float literal.
     Float {
+        /// The value of the float.
         val: f64,
+        /// Trivia data for the Float.
         meta: Trivia<'a>,
+        /// The original string representation of the value.
         raw: &'a str,
     },
-    Bool { val: bool, meta: Trivia<'a> },
+    /// A bool literal.
+    Bool { 
+        /// The value of the boolean.
+        val: bool, 
+        /// Trivia data for the boolean.
+        meta: Trivia<'a> 
+    },
+    /// A datetime literal.
     DateTime {
+        /// The value of the date/time.
         val: ChronoDateTime<FixedOffset>,
+        /// The original string representation of the value.
         raw: &'a str,
+        /// Trivia data for the datetime value.
         meta: Trivia<'a>,
     },
+    /// An array literal.
     Array {
+        /// The contents of the array.
         val: Vec<Item<'a>>,
+        /// Trivia data for the array.
         meta: Trivia<'a>,
     },
+    /// A table literal.
     Table {
+        /// `true` if the table is an array.

```

Comment:
Could this be "`true` if the table is an AoT member"? 
The flag isn't perfectly accurate, we may want to change it to is_aot_member or something similar